cmd/main.go

```diff
@@ -12,114 +13,125 @@ import (
 	"syscall"
 
 	coap "github.com/absmach/coap-cli/coap"
-	"github.com/plgd-dev/go-coap/v2/message"
-	coapmsg "github.com/plgd-dev/go-coap/v2/message"
-	"github.com/plgd-dev/go-coap/v2/message/codes"
-	"github.com/plgd-dev/go-coap/v2/udp/message/pool"
-)
-
-const (
-	get    = "GET"
-	put    = "PUT"
-	post   = "POST"
-	delete = "DELETE"
+	coapmsg "github.com/plgd-dev/go-coap/v3/message"
+	"github.com/plgd-dev/go-coap/v3/message/codes"
+	"github.com/plgd-dev/go-coap/v3/message/pool"
+	"github.com/spf13/cobra"
 )
 
-const (
-	helpCmd = `Use "coap-cli --help" for help.`
-	helpMsg = `
-Usage: coap-cli <method> <URL> [options]
-mathod: get, put, post or delete
--o      observe   option - only valid with GET request (default: false)
--auth   auth option sent as URI Query                  (default: "")
--h      host                                           (default: "localhost")
--p      port                                           (default: "5683")
--d      data to be sent in POST or PUT                 (default: "")
--cf     content format                                 (default: 50 - JSON format))
-
-Examples:
-coap-cli get channels/0bb5ba61-a66e-4972-bab6-26f19962678f/messages/subtopic -auth 1e1017e6-dee7-45b4-8a13-00e6afeb66eb -o
-coap-cli post channels/0bb5ba61-a66e-4972-bab6-26f19962678f/messages/subtopic -auth 1e1017e6-dee7-45b4-8a13-00e6afeb66eb -d "hello world"
-coap-cli post channels/0bb5ba61-a66e-4972-bab6-26f19962678f/messages/subtopic -auth 1e1017e6-dee7-45b4-8a13-00e6afeb66eb -d "hello world" -h 0.0.0.0 -p 1234
-`
+var (
+	host          string
+	port          string
+	contentFormat int
+	auth          string
+	observe       bool
+	data          string
 )
 
-func parseCode(code string) (codes.Code, error) {
-	switch code {
-	case get:
-		return codes.GET, nil
-	case put:
-		return codes.PUT, nil
-	case post:
-		return codes.POST, nil
-	case delete:
-		return codes.DELETE, nil
-	}
-	return 0, errors.New("Message can be GET, POST, PUT or DELETE")
-}
-
 func printMsg(m *pool.Message) {
 	if m != nil {
 		log.Printf("\nMESSAGE:\n %v", m)
 	}
 }
```

Comment:
We can print message f
```suggestion
func printMsg(m *pool.Message) {
	if m != nil {
		log.Printf("\nMESSAGE:\n %s", m.String())
	}
	body, err := m.ReadBody()
	if err != nil {
		log.Fatalf("failed to read body %s", err.Error())
	}
	log.Printf("MESSAGE BODY: %s\n", string(body))
}
```